src/shared/hooks/useLaunchDarkly.ts

```diff
@@ -0,0 +1,56 @@
+import { LDFlagValue, useFlags, useLDClient } from 'launchdarkly-react-client-sdk';
+
+import { FeatureSegments, FeatureFlags, FeatureFlagsKeys } from 'shared/types/featureFlags';
+
+import { PROHIBITED_PII_KEYS } from './useLaunchDarkly.const';
+
+/**
+ * Internal wrapper for LaunchDarkly's hooks and flags.
+ */
+export const useLaunchDarkly = () => {
+  const ldClient = useLDClient();
+  const flags = useFlags();
+
+  const identifyLDContext = (
+    context: {
+      userId: string;
+      workspaceId?: string;
+      appletId?: string;
+      featureTests?: FeatureSegments[];
+    },
+    onDone?: ((err: Error | null) => void) | undefined,
+  ) => {
+    if (PROHIBITED_PII_KEYS.some((val) => Object.keys(context).includes(val))) {
```

Comment:
It might make more sense to have an allowlist (`workspaceId`, `appletId`, `featureTests`) instead of blocklist. It seems unlikely we'd be able to predict the keys that someone might pass that could be problematic

Reply:
I think we still don't know _what_ we're going to be sending to properly identify a user's context...but point taken, I'll go with allow list instead

Reply:
Yes that's very true! So all TBD...